src/Products/Viewer/Cache/PngViewer.cs

```diff
@@ -0,0 +1,187 @@
+﻿using GroupDocs.Viewer.Options;
+using GroupDocs.Viewer.Results;
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace GroupDocs.Viewer.MVC.Products.Viewer.Cache
+{
+    class PngViewer : IDisposable, ICustomViewer
+    {
+        private readonly string filePath;
+        private readonly string htmlResourcePrefix;
+        private readonly IViewerCache cache;
+
+        private readonly GroupDocs.Viewer.Viewer viewer;
+        private readonly PngViewOptions pngViewOptions;
+        private readonly ViewInfoOptions viewInfoOptions;
+        private static readonly Common.Config.GlobalConfiguration globalConfiguration = new Common.Config.GlobalConfiguration();
+
+        public PngViewer(string filePath, string htmlResourcePrefix, IViewerCache cache, LoadOptions loadOptions, int pageNumber = -1, int newAngle = 0)
+        {
+            this.cache = cache;
+            this.filePath = filePath;
+            this.htmlResourcePrefix = htmlResourcePrefix;
+            this.viewer = new GroupDocs.Viewer.Viewer(filePath, loadOptions);
+            this.pngViewOptions = this.CreatePngViewOptions(pageNumber, newAngle);
+            this.viewInfoOptions = ViewInfoOptions.FromPngViewOptions(this.pngViewOptions);
+        }
+
+        public GroupDocs.Viewer.Viewer GetViewer()
+        {
+            return this.viewer;
+        }
+
+        private PngViewOptions CreatePngViewOptions(int passedPageNumber = -1, int newAngle = 0)
+        {
+            PngViewOptions pngViewOptions = new PngViewOptions(pageNumber =>
+            {
+                string fileName = $"p{pageNumber}.png";
+                string filePath = this.cache.GetCacheFilePath(fileName);
```

Comment:
Codacy found an issue: [Rename 'filePath' which hides the field with the same name.](https://app.codacy.com/manual/GroupDocs/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5933706)